js/src/common/states/ModalManagerState.js

```diff
@@ -0,0 +1,43 @@
+import Modal from '../components/Modal';
+
+export default class ModalManagerState {
+  constructor() {
+    this.modal = null;
+  }
+
+  /**
+   * Show a modal dialog.
+   *
+   * @public
+   */
+  show(type, attrs) {
```

Comment:
If you agree with my comment about renaming `type` to `component` or `componentClass` in https://github.com/flarum/core/pull/2163#discussion_r446678767 I think we should use an identical naming convention here.

Reply:
I assume the parameters here are switched because `type` in AlertManager#show is optional, right?

Reply:
I noticed that as well. Here, `attrs` is optional. Whether with Alerts, `type` is optional.

So I think switching them makes the most sense.

Reply:
I don't understand your suggestion here, @clarkwinkelmann, and your response seems to be missing a word. :thinking: 

Are you saying that having them switched (when compared with alerts) makes sense, or are you suggesting to switch them again?

Reply:
Sorry for the confusion. I think the current version (switched) is good. We place the required arguments first and optional arguments second.

The requested change of this thread was only to rename, not reorder.